sklearn/preprocessing/tests/test_target_encoder.py

```diff
@@ -70,26 +71,26 @@ def test_encoding(categories, unknown_value, global_random_seed, smooth, target_
 
     if categories == "auto":
         X_train = X_train_array
-    else:
-        X_train = categories[0][X_train_array]
-
-    if categories == "auto":
         X_test = X_test_array
     else:
+        X_train = categories[0][X_train_array]
         X_test = categories[0][X_test_array]
+
     X_test = np.concatenate((X_test, [[unknown_value]]))
 
     rng = np.random.RandomState(global_random_seed)
 
+    n_splits = 3
```

Comment:
Can you revert the changes to `test_encoding` + typos changes and let https://github.com/scikit-learn/scikit-learn/pull/26958 handle it?

Reply:
Sorry, was hoping #26958 would be a quick merge and then I could just rebase. Alas, will revert!